src/security/decorators.py

```diff
@@ -931,6 +1174,27 @@ def wrapper(request, *args, **kwargs):
 
     return wrapper
 
+
+def article_stage_review_required(func):
+    """
+    Checks that the article is in one of the review stages
+    :param func: func
+    :return: PermissionDenied or func
+    """
+    def review_required_wrapper(request, *args, **kwargs):
+        article = get_object_or_404(
+            models.Article,
+            pk=kwargs['article_id']
```

Comment:
Could throw a KeyError.

Reply:
it would 404 before a key error.

Reply:
Definately not, `__getitem__` will be invoked on kwargs before `get_object_or_404` is called

Reply:
handler and test added